packages/web/src/components/pages/TestSuites/TestSuiteDetails/TestSuiteSettings/SettingsGeneral/SettingsGeneral.tsx

```diff
@@ -15,17 +13,16 @@ import {useDeleteTestSuiteMutation, useUpdateTestSuiteMutation} from '@services/
 import {useEntityDetailsPick} from '@store/entityDetails';
 
 const SettingsGeneral: FC = () => {
-  const isWritable = useSystemAccess(SystemAccess.agent);
   const mayDelete = usePermission(Permissions.deleteEntity);
   const {details} = useEntityDetailsPick('details');
   const deleteTestSuiteExtension = usePluginSlot('deleteTestSuiteExtension');
   const [deleteTestSuite] = useDeleteTestSuiteMutation();
 
   return (
     <>
-      <NameNDescription label="test suite" useUpdateEntity={useUpdateTestSuiteMutation} readOnly={!isWritable} />
-      <Labels label="test suite" useUpdateEntity={useUpdateTestSuiteMutation} readOnly={!isWritable} />
-      {mayDelete && isWritable ? (
+      <NameNDescription label="test suite" useUpdateEntity={useUpdateTestSuiteMutation} readOnly={details.readOnly} />
```

Comment:
Not sure if important: Probably it should take in account also the `isWritable` from before - otherwise, there is a gap when the Test Suite is read-only, but it's not perceived as read-only yet

Reply:
this will not work for mixed case, where the agent is online, but we have a read-only test-suite which is not present on the agent anymore